lib/workload/stateful/database/component.ts

```diff
@@ -1,86 +1,96 @@
 import { Construct } from 'constructs';
-import { Aspects, RemovalPolicy } from 'aws-cdk-lib';
+import { RemovalPolicy, Duration } from 'aws-cdk-lib';
 import * as rds from 'aws-cdk-lib/aws-rds';
 import * as ec2 from 'aws-cdk-lib/aws-ec2';
 
-export interface DatabaseProps {
+/**
+ * Props for enabling enhanced monitoring.
+ */
+type MonitoringProps = {
+  /**
+   * Add cloud watch exports.
+   */
+  readonly cloudwatchLogsExports?: string[];
+  /**
+   * Enable performance insights.
+   */
+  readonly enablePerformanceInsights?: boolean;
+  /**
+   * performance insights retention period
+   */
+  readonly performanceInsightsRetention?: rds.PerformanceInsightRetention;
+  /**
+   * Enable enhanced monitoring by specifying the interval
+   */
+  readonly enhancedMonitoringInterval?: Duration;
+};
+
+export type DatabaseProps = MonitoringProps & {
   clusterIdentifier: string;
   defaultDatabaseName: string;
   parameterGroupName: string;
   username: string;
-  version: rds.AuroraMysqlEngineVersion;
+  version: rds.AuroraPostgresEngineVersion;
   numberOfInstance: number;
   minACU: number;
   maxACU: number;
-  allowDbSGIngressRule?: {
-    peer: ec2.IPeer;
-    description?: string;
-  }[];
-}
+  dbPort: number;
+  allowedInboundSG?: ec2.SecurityGroup;
+};
 
 export class DatabaseConstruct extends Construct {
   readonly dbSecurityGroup: ec2.SecurityGroup;
   readonly dbCluster: rds.DatabaseCluster;
 
   constructor(scope: Construct, id: string, vpc: ec2.IVpc, props: DatabaseProps) {
     super(scope, id);
-    const dbPort = 3306;
 
-    const secret = new rds.DatabaseSecret(this, id + 'Secret', {
+    const dbSecret = new rds.DatabaseSecret(this, id + 'DbSecret', {
```

Comment:
Is the idea that every microservice that needs the database will fetch this secret to connect?

Reply:
I am thinking this way but do you have any suggestion?

Realising this, I should have pass in the secret name so that it could be discovered by the microservice.

Reply:
I think this is okay. Probably simplest to just have one secret for the whole database cluster.

Reply:
Ha, I was thinking similarly for the DB stack to set up databases / users for the different services and isolate them against each other... but let's start simple ;-)